hoot-core-test/src/test/cpp/hoot/core/io/OsmApiWriterTestServer.cpp

```diff
@@ -99,35 +102,145 @@ bool RetryConflictsTestServer::respond(HttpConnection::HttpConnectionPtr& connec
+  //  Create changeset
+  else if (headers.find(OsmApiWriter::API_PATH_CREATE_CHANGESET) != std::string::npos)
+    message = HTTP_200_OK + "1\r\n";
+  //  Upload changeset 1
+  else if (headers.find("/api/0.6/changeset/1/upload/") != std::string::npos)
+  {
+    //  The first time through, the 'version' of element 1 should fail.
+    if (!_has_error)
+    {
+      message = HTTP_409_CONFLICT + "Changeset conflict: Version mismatch: Provided 2, server had: 1 of Way 1";
+      _has_error = true;
+    }
+    else
+      message = HTTP_200_OK + OsmApiSampleRequestResponse::SAMPLE_CHANGESET_1_RESPONSE + "\r\n";
+  }
+  //  Get way 1's updated version
+  else if (headers.find("/api/0.6/way/1") != std::string::npos)
+    message = HTTP_200_OK + OsmApiSampleRequestResponse::SAMPLE_ELEMENT_1_GET_RESPONSE + "\r\n";
+  //  Close changeset
+  else if (headers.find("/close/"))
+  {
+    message = HTTP_200_OK;
+    continue_processing = false;
+  }
+  else
+  {
+    //  Error out here
+    message = HTTP_404_NOT_FOUND;
+    continue_processing = false;
+  }
+  //  Write out the response
+  write_response(connection, message);
+  //  Return true if we should continue listening and processing requests
+  return continue_processing;
+}
+
+const char* OsmApiSampleRequestResponse::SAMPLE_CAPABILITIES_RESPONSE =
+    "<?xml version='1.0' encoding='UTF-8'?>\n"
+    "<osm version='0.6' generator='OpenStreetMap server'>\n"
+    "  <api>\n"
+    "    <version minimum='0.6' maximum='0.6'/>\n"
+    "    <area maximum='0.25'/>\n"
+    "    <note_area maximum='25'/>\n"
+    "    <tracepoints per_page='5000'/>\n"
+    "    <waynodes maximum='2000'/>\n"
+    "    <changesets maximum_elements='10000'/>\n"
+    "    <timeout seconds='300'/>\n"
+    "    <status database='online' api='online' gpx='online'/>\n"
+    "  </api>\n"
+    "  <policy>\n"
+    "    <imagery>\n"
+    "      <blacklist regex='http://xdworld\\.vworld\\.kr:8080/.*'/>\n"
+    "      <blacklist regex='.*\\.here\\.com[/:].*'/>\n"
+    "    </imagery>\n"
+    "  </policy>\n"
     "</osm>";
-const char* OsmApiSampleResponses::SAMPLE_PERMISSIONS =
-    "<?xml version='1.0' encoding='UTF-8'?>"
-    "<osm version='0.6' generator='OpenStreetMap server'>"
-    "  <permissions>"
-    "    <permission name='allow_read_prefs'/>"
-    "    <permission name='allow_read_gpx'/>"
-    "    <permission name='allow_write_api'/>"
-    "    <permission name='allow_write_gpx'/>"
-    "  </permissions>"
+const char* OsmApiSampleRequestResponse::SAMPLE_PERMISSIONS_RESPONSE =
+    "<?xml version='1.0' encoding='UTF-8'?>\n"
+    "<osm version='0.6' generator='OpenStreetMap server'>\n"
+    "  <permissions>\n"
+    "    <permission name='allow_read_prefs'/>\n"
+    "    <permission name='allow_read_gpx'/>\n"
+    "    <permission name='allow_write_api'/>\n"
+    "    <permission name='allow_write_gpx'/>\n"
+    "  </permissions>\n"
+    "</osm>";
```

Comment:
Added `\n` at the end of all of the sample responses to make `--trace` output easier to read.